packages/wagmi/src/hooks/useWaitForMessageIdentifier.ts

```diff
@@ -0,0 +1,31 @@
+import {
+  extractMessageIdentifierFromLogs,
+  type MessageIdentifier,
+} from '@eth-optimism/viem'
+import { useMemo } from 'react'
+import type { Hash, PublicClient } from 'viem'
+import { useConfig, usePublicClient, useWaitForTransactionReceipt } from 'wagmi'
+
+export type WaitForMessageIdentifierParams = {
+  hash: Hash
+}
+
+export const useWaitForMessageIdentifier = ({
+  hash,
+}: WaitForMessageIdentifierParams) => {
+  const config = useConfig()
+  const publicClient = usePublicClient({ config })
+  const { data } = useWaitForTransactionReceipt({ config, hash })
+
+  const id = useMemo<Promise<MessageIdentifier | undefined>>(async () => {
+    if (!data) {
+      return
+    }
+    return await extractMessageIdentifierFromLogs(
```

Comment:
doesnt have to be done in this PR but, `extractMessageIdentifierFromLogs` will need to be updated now that L2toL2CrossDomainMessenger is not logging with the SentMessage event.

Reply:
I also, wonder if we should extend `extractMessageIdentifierFromLogs` to also return the message payload, since that is also needed in order to execute a message.

Reply:
yea that was actually the intention to include the payload as well! i'll update it in the next PR i do